vsintegration/src/FSharp.Editor/LanguageService/ProjectSitesAndFiles.fs

```diff
@@ -274,6 +274,7 @@ type internal ProjectSitesAndFiles() =
         let option =
             let newOption () = {
                 ProjectFileName = projectSite.ProjectFileName
+                ProjectId = projectId |> Option.map (fun x -> x.Id.ToString())
```

Comment:
Not sure if this matters, but the `ToString` there is culture aware, so might return different results in different cultures (or if the thread's current culture changes between calls, or if calls happen on different threads that have different cultures).

Reply:
That's a good point, I need to fix that.

Reply:
In a recent change, I do the comparison with `StringComparison.OrdinalIgnoreCase`.

Reply:
shouldn't it follow same pattern as above `let strGuid = "_ProjectId=" + (match options.ProjectId with Some(projectId) -> projectId | _ -> "null")` - mabe introduce a helper?

Reply:
I was actually intending to specify a specific formatting (and possibly invariant culture) to both the ToString and the comparison.

Just `OrdinalIgnoreCase` is not the correct fix.

I also agree with @forki about extracting a helper to generated this string.

Reply:
The `Id` is a `Guid`. Is it not safe to just simply `ToString()` it? Its default format is "D". What else would we need to do it?

I'm thinking about changing `FSharpProjectOptions`'s `ProjectId` field to just be a `Guid option` so we can definitely can say we will never worry about this. The downside is that, any other tool (ionide, rider, vs4mac) that integrates with FCS may have to do their own mapping if they already do not represent projects as a `Guid`.

Reply:
I think you are right that it's fine according to https://msdn.microsoft.com/en-us/library/97af8hh4(v=vs.110).aspx.  I just don't like unspecified formats used for this sort of thing, since they go wrong in too many instances.

Reply:
In that case, it may be better to make the field a `Guid` instead of a string so it's clear. What do you think?

Reply:
I'm thinking it would force client to maintain a mapping between such guid and project name / target couples.

A simpler way for the client would be to concatenate the target name to the project name and use that as projectid, but this is only possible when ProjectId is a string.

Reply:
After some discussion, I will make the id case sensitive. Then I will be explicit with the `ToString()` to be the "D" format and make it lowercase.

Reply:
👍